core/src/main/java/com/brandwatch/robots/RobotsServiceImpl.java

```diff
@@ -70,10 +78,19 @@ public boolean isAllowed(@Nonnull String crawlerAgentString, @Nonnull URI resour
         final URI robotsUri = utilities.getRobotsURIForResource(resourceUri);
         log.debug("Resolved robots URI to: {}", robotsUri);
 
+        String host = robotsUri.getHost();
+        if (host != null && excludedDomains.stream().anyMatch(host::contains)) {
```

Comment:
Seems inefficient to stream all entries vs just doing `excludedDomains.contains(host)` .. oh, are you doing arbitrary substring matching?

Reply:
I've updated it to use a more clear matching behaviour as suggested (i.e.`equals(domain) ||  endsWith(".domain")`)